examples/process-manager-plugin/supervisor/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ligato/cn-infra/agent"
+
+	bs "github.com/ligato/cn-infra/exec/supervisor"
+	"github.com/ligato/cn-infra/logging"
+)
+
+func main() {
+	// The supervisor plugin defines a configuration file allowing to easily manage processes using process
+	// manager plugin.
+	//
+	// The config file can be put to the supervisor either via flag "-supervisor-config="
+	// or define its path in the environment variable "SUPERVISOR_CONFIG". Another option is
+	// to define config directly with "UseConf()" - this option will be shown in the example. A sample of
+	// YAML config file can be found in the processes/supervisor folder.
+
+	log := logging.DefaultLogger
+
+	// Define three processes p1-p3 where the third process is set to terminate p2 if stopped
+	conf := bs.Config{
+		Processes: []bs.Process{
+			{
+				Name:        "p1",
+				LogFilePath: "example.log",
+				BinaryPath:  "../test-process/test-process",
+				Args:        []string{"-max-uptime=60"},
+			},
+			{
+				Name:        "p2",
+				LogFilePath: "example.log",
+				BinaryPath:  "../test-process/test-process",
+			},
+			{
+				Name:        "p3",
+				LogFilePath: "example.log",
+				BinaryPath:  "../test-process/test-process",
+				Required:    true,
+			},
+		},
+	}
+
+	// start plugin
+	bsp := bs.NewPlugin(bs.UseConf(conf))
+
+	a := agent.NewAgent(agent.AllPlugins(bsp))
+	if err := a.Start(); err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := a.Stop(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// give the agent time to start
+	time.Sleep(5 * time.Second)
```

Comment:
This should not be needed. The `agent.Start()` will return only after all plugins are successfully initialized.